src/main/jbake/content/blog/2016/05/04/tags-and-string-metrics.adoc

```diff
@@ -0,0 +1,135 @@
+== Tags
+Hawkular Metrics has had support for tags for a while now. Tags come in the form of key/value pairs, and they can be
+applied to the metric (also commonly referred to as metric definition or time series). Tags can now also be added to
+individual data points. Let's first review how tags can be used up until now.
+
+=== Metric Tags
+Tags can be supplied when creating a metric.
+
+[source,shell]
+.Create metric with tags
+----
+curl -X POST http://server/hawkular/metrics/gauges -d @request.json \
+-H "Content-Type: application/json" -H "Hawkular-Tenant: com.acme"
+----
+
+[source,javascript]
+.request.json
+----
+{
+  "id": "request_size",
+  "tags": {
+    "datacenter": "dc1",
+    "env": "stage"
+    "units": "bytes"
+  }
+}
+----
+
+This request creates a metric named request_size that has three tags. Tags can be added, updated, and removed. Tags can
+also be used in queries for filtering and grouping data across multiple metrics. See the
+link:http://www.hawkular.org/docs/components/metrics/index.html#_tagging[user guide] for examples.
+
+=== Data Point Tags
+The 0.15.0 release introduces support for data point tags. Tags can be supplied when adding data points.
+
+[source,shell]
+.Add data points with tags
+----
+curl -X POST http://server/hawkular/metrics/gauges/request_size -d @request.json \
+-H "Content-Type: application/json" -H "Hawkular-Tenant: com.acme"
+----
+
+[source,javascript]
+.request.json
+----
+[
+  {
+    "timestamp": 12345,
+    "value": 3.14
+    "tags": {
+      "hostname": "server01"
+    }
+  },
+  {
+    "timestamp": 56789,
+    "value": 3.14
+    "tags": {
+      "hostname": "server02"
+    }
+  },
+  {
+    "timestamp": 90123,
+    "value": 3.14
+    "tags": {
+      "hostname": "server03"
+    }
+  }
+]
+----
+
+Data points, including their tags, are immutable. Unlike metric tags data point tags cannot be modified, nor can they
+be deleted. This is an important distinction. Data point tags can be used to group data points when querying a single
+metric.
+
+=== Which One Should I Use?
+At this point you might be asking when should I use metric tags instead of data point tags and vice versa. As often the
+case, some examples help answer these types of questions. Let's use the request_size metric from the previous examples.
+We want to add a hostname tag. The source of this metric is an HTTP server, and we want to monitor the overall request
+sizes it handles over some period of time. The data points will always come from the same host; consequently, a metric
+tag should be used in this case.
+
+Now suppose the request_size metric is tracking the request size across multiple HTTP servers, each of which is running
+on a separate host. We want the ability to correlate data points with the host from which they come. In this instance,
+the hostname tag should be supplied with each data point.
+
+Let's take the example of tracking request size across multiple servers one step further. We also want to know in which
```

Comment:
The idea came of APIMan's use cases in which that APIMan needs to track the client for each request. Each request can come from any number of clients. Maybe I should change the example to be along these lines.